File: shared/src/commonMain/kotlin/com/mobile/fairless/common/navigation/Navigator.kt
package com.mobile.fairless.common.navigation

import com.mobile.fairless.features.main.models.Shop


enum class ScreenRoute(val isMain: Boolean) {
    //NavBar
    Main(true),
    Notification(true),
    Message(true),
    Menu(true),
    Search(true),

    Shop(false),
    Document(false),
    Profile(false),
    Auth(false),
    AboutFairLess(false),
    Rules(false),
    Feedback(false),
    Faq(false),
    AboutApp(false)
}

interface Navigator {
    fun navigateBack()

    fun navigateToMain()
    fun navigateToSearch()
    fun navigateToProfile()
    fun navigateToNotification()
    fun navigateToMessage()
    fun navigateToMenu()
    fun navigateToAuth()
    fun navigateToDocument(product: String)
    fun navigateToAboutFairLess()
    fun navigateToRules()
    fun navigateToFeedback()
    fun navigateToFaq()
    fun navigateToAboutApp()
    fun navigateToShop(shop: String)
}

File: iosApp/iosApp/navigation/Navigator.swift
//
//  Navigator.swift
//  iosApp
//
//  Created by Егор Лядский on 2023-08-08.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation
import SwiftUI
import shared

@MainActor class NavigatorImpl: UIViewController, Navigator, ObservableObject {
    
    @Environment(\.appRoutingState) var appRoutingState: AppRoutingState
    
    static let shared = NavigatorImpl()
    let startDestination = ScreenRoute.main
    
    nonisolated func navigateBack() {
    }
    
    nonisolated func navigateToAboutApp() {
        
    }
    
    nonisolated func navigateToAboutFairLess() {
        
    }
    
    nonisolated func navigateToAuth() {
        
    }
    
    nonisolated func navigateToDocument(product: String) {
        
    }
    
    nonisolated func navigateToFaq() {
        
    }
    
    nonisolated func navigateToFeedback() {
        
    }
    
    func navigateToMain() {
        appRoutingState.currentScreen = .main
    }
    
    func navigateToMenu() {
        appRoutingState.currentScreen = .menu
    }
    
    nonisolated func navigateToMessage() {
        
    }
    
    nonisolated func navigateToNotification() {
        
    }
    
    nonisolated func navigateToProfile() {
        
    }
    
    nonisolated func navigateToRules() {
        
    }
    
    nonisolated func navigateToSearch() {
        
    }
    
    nonisolated func navigateToShop(shop_ shop: String) {
        
    }
}
